src/mpb.cpp

```diff
@@ -591,6 +600,28 @@ void *fields::get_eigenmode(double frequency, direction d, const volume where, c
     } while (match_frequency &&
              fabs(sqrt(eigvals[band_num - 1]) - frequency) > frequency * match_tol);
 
+  if (am_master() && dp) {
+    scalar_complex s = {real(dp->get_s()),imag(dp->get_s())};
+    scalar_complex p = {real(dp->get_p()),imag(dp->get_p())};
+
+    master_printf("diffracted planewave: k = %g, %g, %g for omega = %g\n", k[0],k[1],k[2], frequency);
+    update_maxwell_data_k(mdata, k, G[0], G[1], G[2]);
+    maxwell_set_planewave(mdata, H, band_num, dp->get_g(), s, p, dp->get_axis());
+
+    eigvals[band_num - 1] = frequency;
```

Comment:
or if match_frequency is false then it needs to compute the `frequency` from `k` and `get_g()`.